product_variant_supplierinfo/views/product_supplierinfo_view.xml

```diff
@@ -5,10 +5,8 @@
     <field name="inherit_id" ref="product.product_supplierinfo_form_view"/>
     <field name="model">product.supplierinfo</field>
     <field name="arch" type="xml">
-        <field name="product_name" position="before">
-            <field name="product_id"
-                   domain="[('product_tmpl_id', '=', parent.id)]"
-                   invisible="context.get('hide_product_id')"/>
+        <field name="product_id" position="attributes">
+            <attribute name="invisible">context.get('hide_product_id')</attribute>
```

Comment:
why did you remove the domain?

Reply:
Domain is inherited from father view.
From product.product_view:
`<field name="product_id" domain="[('product_tmpl_id', '=', product_tmpl_id)]" invisible="1"/>`

Reply:
Ok, thanks!